lucetc/src/lib.rs

```diff
@@ -31,8 +31,13 @@ use std::env;
 use std::path::{Path, PathBuf};
 use tempfile;
 
+enum LucetcInput {
+    Bytes(Vec<u8>),
+    Path(PathBuf),
+}
```

Comment:
We have a `LucetcInput` enum rather than reading the bytes in `new`, and changing the type of `input` to a `Vec<u8>` because we don't always have a program _in_ the given path when `Lucetc` is constructed. (The `lucet-wasi-sdk/src/lib.rs:compile_to_lucet` test is an example of this)

Instead, we use a lil' enum so that the order of operations doesn't need to change, and can handle when we are directly given bytes, or when we have been given a path from which to read later.